Bio/Align/clustal.py

```diff
@@ -9,94 +9,100 @@
 You are expected to use this module via the Bio.Align functions (or the
 Bio.SeqIO functions if you are interested in the sequences only).
 """
+import Bio
 from Bio.Align import Alignment
+from Bio.Align import interfaces
 from Bio.Seq import Seq
 from Bio.SeqRecord import SeqRecord
 
 
-class ClustalWriter:
+class AlignmentWriter(interfaces.AlignmentWriter):
     """Clustalw alignment writer."""
 
-    def write_alignment(self, alignment):
-        """Use this to write (another) single alignment to an open file."""
-        if len(alignment) == 0:
-            raise ValueError("Must have at least one sequence")
-        if alignment.get_alignment_length() == 0:
-            # This doubles as a check for an alignment object
-            raise ValueError("Non-empty sequences are required")
-
+    def write_header(self, alignments):
+        """Use this to write the file header."""
+        stream = self.stream
         try:
-            version = str(alignment._version)
+            program = alignments.program
         except AttributeError:
-            version = ""
-        if not version:
-            version = "1.81"
-        if version.startswith("2."):
-            # e.g. 2.0.x
-            output = "CLUSTAL %s multiple sequence alignment\n\n\n" % version
+            program = "Biopython"
+            version = Bio.__version__
         else:
-            # e.g. 1.81 or 1.83
-            output = "CLUSTAL X (%s) multiple sequence alignment\n\n\n" % version
+            try:
+                version = alignments.version
+            except AttributeError:
+                version = ""
+        line = "%s %s multiple sequence alignment\n" % (program, version)
```

Comment:
It has to start with CLUSTAL for many tools to accept it. See for example the MUSCLE command line tool which offers Clustal style output but with MUSCLE here, and strict Clustal output with the magic word MUSCLE.

See also our own parser:
https://github.com/biopython/biopython/blob/biopython-179/Bio/AlignIO/ClustalIO.py#L113

Reply:
Are you sure? Most alignment tools that produce output files in the CLUSTAL format use their own program name here:
```
$ head -1 Clustalw/clustalw.aln
CLUSTAL W (1.81) multiple sequence alignment

$ head -1 Clustalw/msaprobs.aln
MSAPROBS version 0.9.7 multiple sequence alignment

$ head -1 Clustalw/muscle.aln
MUSCLE (3.8) multiple sequence alignment

$ head -1 Clustalw/kalign.aln
Kalign (2.0) alignment in ClustalW format

$ head -1 Clustalw/probcons.aln
PROBCONS version 1.12 multiple sequence alignment
```
The exception is PROMALS3D, which uses
```
$ head -1 Clustalw/promals3d.aln 
CLUSTAL format multiple sequence alignment by PROMALS3D
```

Reply:
Maybe the consensus has shifted in the last 10 years, at least most include the keywords "multiple sequence alignment" now.

Reply:
I don't have a strong opinion about this, but perhaps it is better to avoid using the word `CLUSTAL` here if the alignment was not actually generated by clustal.

Reply:
Added Biopython to the whitelist in Bio.AlignIO.ClustalIO, and added a test for it.